Json Format

ibirrer opened the pull request:
Adds `json` graph format as a first step to resolve #32.
Note that options like `showConflicts` and `showDuplicates` have no effect on the output. The json always contains all information,  because it is considered that the consumer of the json files will filter the data as needed. 
For the same reason, the aggregated graph contains all dependencies, including conflicts.

ferstl commented:
@ibirrer 
Thanks for the PR! Before merging, I'd like to discuss the showConflicts/showDuplicates options in the aggregated graph:
> Note that options like showConflicts and showDuplicates have no effect on the output. The json always contains all information, because it is considered that the consumer of the json files will filter the data as needed.

I don't think this is a good idea. As I explained in the [FAQ entry](https://github.com/ferstl/depgraph-maven-plugin) _"Why can't I show duplicates and conflicts in the aggregated graph?"_ (at the bottom of the page), this could lead to very confusing results. 
The reason is that the aggregated graph is the "sum" of all modules' individual graphs and each module might see the same dependency from `A` to `B` in different resolutions (included/duplicate/conflict). So in the graphical representation there might be two nodes with three arrows between them. But you won't see which module caused the duplicate or conflict.

Note, that the aggregated graph is not a dependency tree. If several modules have a dependency (direct or transitive) to `X`, the graph will show all different paths to that dependency.

ibirrer commented:
Thanks for the comment. I understand your points about the aggregated graph. I reverted the change in the `AggregatingDependencyGraphMojo` (and also rebased to your latest master). I still think that the json should always show group id, version, conflict and duplicates for the normal graph and changed `DependencyGraphMojo` accordingly.

The output of `mvn depgraph:graph -DgraphFormat=json` of module-3 now looks as follows:

```
{ "artifacts":
  [ { "id": 0
    , "artifactId": "module-1"
    , "groupId": "com.github.ferstl"
    , "version": "1.0.0-SNAPSHOT"
    }
  , { "id": 1
    , "artifactId": "commons-codec"
    , "groupId": "commons-codec"
    , "version": "1.10"
    }
  , { "id": 2
    , "artifactId": "commons-lang3"
    , "groupId": "org.apache.commons"
    , "version": "3.1"
    }
  , { "id": 3
    , "artifactId": "module-3"
    , "groupId": "com.github.ferstl"
    , "version": "1.0.0-SNAPSHOT"
    }
  , { "id": 4
    , "artifactId": "module-2"
    , "groupId": "com.github.ferstl"
    , "version": "1.0.0-SNAPSHOT"
    }
  , { "id": 5
    , "artifactId": "guava"
    , "groupId": "com.google.guava"
    , "version": "21.0"
    }
  , { "id": 6
    , "artifactId": "commons-lang3"
    , "groupId": "org.apache.commons"
    , "version": "3.0"
    }
  , { "id": 7
    , "artifactId": "querydsl-core"
    , "groupId": "com.mysema.querydsl"
    , "version": "3.7.4"
    }
  , { "id": 8
    , "artifactId": "jsr305"
    , "groupId": "com.google.code.findbugs"
    , "version": "1.3.9"
    }
  , { "id": 9
    , "artifactId": "mysema-commons-lang"
    , "groupId": "com.mysema.commons"
    , "version": "0.2.4"
    }
  , { "id": 10
    , "artifactId": "bridge-method-annotation"
    , "groupId": "com.infradna.tool"
    , "version": "1.13"
    }
  ]
, "dependencies":
  [ { "from": 0
    , "to": 1
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 0
    , "to": 2
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 3
    , "to": 0
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 4
    , "to": 0
    , "resolution": "OMITTED_FOR_DUPLICATE"
    , "scopes": [ "compile" ]
    }
  , { "from": 4
    , "to": 5
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 4
    , "to": 1
    , "resolution": "OMITTED_FOR_DUPLICATE"
    , "scopes": [ "compile" ]
    }
  , { "from": 4
    , "to": 6
    , "resolution": "OMITTED_FOR_CONFLICT"
    , "scopes": [ "compile" ]
    }
  , { "from": 3
    , "to": 4
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 7
    , "to": 5
    , "resolution": "OMITTED_FOR_DUPLICATE"
    , "scopes": [ "compile" ]
    }
  , { "from": 7
    , "to": 8
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 7
    , "to": 9
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 7
    , "to": 10
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  , { "from": 3
    , "to": 7
    , "resolution": "INCLUDED"
    , "scopes": [ "compile" ]
    }
  ]
}
```

ferstl commented:
I polished the implementation a bit and made some slight adjustments to the JSON format:
https://github.com/ferstl/depgraph-maven-plugin/pull/47 

OK?
